pkg/reconciler/cronjobsource/cronjobsource.go

```diff
@@ -155,18 +157,35 @@ func (r *Reconciler) reconcile(ctx context.Context, cronjob *v1alpha1.CronJobSou
 
 	ra, err := r.createReceiveAdapter(ctx, cronjob, sinkURI)
 	if err != nil {
-		r.Logger.Error("Unable to create the receive adapter", zap.Error(err))
+		logging.FromContext(ctx).Error("Unable to create the receive adapter", zap.Error(err))
 		return fmt.Errorf("creating receive adapter: %v", err)
 	}
 	cronjob.Status.PropagateDeploymentAvailability(ra)
 
+	// TODO Delete this after 0.8 is cut.
+	if status.DeploymentIsAvailable(&ra.Status, true) {
+		err = r.deleteOldReceiveAdapter(ctx, cronjob)
+		if err != nil {
+			return fmt.Errorf("deleting old receive adapter: %v", err)
+		}
+	}
 	_, err = r.reconcileEventType(ctx, cronjob)
 	if err != nil {
 		cronjob.Status.MarkNoEventType("EventTypeReconcileFailed", "")
 		return fmt.Errorf("reconciling event types: %v", err)
 	}
 	cronjob.Status.MarkEventType()
 
+	// TODO Delete this after 0.8 is cut.
+	oldEventType, err := r.getOldEventType(ctx, cronjob)
+	if err != nil {
+		return fmt.Errorf("getting old event type: %v", err)
+	} else if oldEventType != nil {
+		if err = r.EventingClientSet.EventingV1alpha1().EventTypes(cronjob.Namespace).Delete(oldEventType.Name, nil); err != nil {
```

Comment:
don't you need an empty DeleteOptions instead of nil? 
I think I've seen it being call like that. If nil doesn't break, I guess is fine

Reply:
I see further down that you use &DeleteOptions{}, let's use that then

Reply:
Done.